superset-frontend/src/components/URLShortLinkButton/index.jsx

```diff
@@ -50,9 +53,20 @@ class URLShortLinkButton extends React.Component {
 
   getCopyUrl(e) {
     e.stopPropagation();
-    getShortUrl(this.props.url)
-      .then(this.onShortUrlSuccess)
-      .catch(this.props.addDangerToast);
+    const nativeFiltersKey = getUrlParam(URL_PARAMS.nativeFiltersKey);
+    if (this.props.dashboardId) {
+      getFilterValue(this.props.dashboardId, nativeFiltersKey)
```

Comment:
Maybe `getFilterValue` should be renamed to `getSavedFilterState` for clarify.

We should probably also just pass the full state to `URLShortLinkButton` itself so we can avoid a round trip to the server for the saved states---because the client states should always be in sync with the URL. If the states already exist somewhere on the client, then we can pass it around for generating the short URL. 

By passing full states (an arbitrary JSON object + a base URL template), it also decouples `URLShortLinkButton` with either dashboard or chart API endpoint, so it can be used in even other places (e.g. SQL queries, etc).